core/decorators/async-setup-decorator.ts

```diff
@@ -5,6 +5,7 @@ import { SETUP } from "./_metadata-keys";
 export function AsyncSetup(target: object,
                            decoratedPropertyKey: string,
                            descriptor?: TypedPropertyDescriptor<() => any>) {
+   descriptor = undefined; // Unused
```

Comment:
Similarly this seems like it's just doing too much I'm not sure I agree with the no unused arguments